src/pointarray.cpp

```diff
@@ -387,10 +506,53 @@ V3d PointArray::pickVertex(const V3d& cameraPos,
                            double longitudinalScale, double* distance,
                            std::string* info) const
 {
+    // NOTE rayOrigin here is not cameraPos but something else
+    // This routine uses cameraPos as the rayOrigin point
+
     if (m_npoints == 0)
         return V3d(0);
-    size_t idx = closestPointToRay(m_P, m_npoints, rayOrigin - offset(),
-                                   rayDirection, longitudinalScale, distance);
+
+    const V3f cameraPosOffset = cameraPos - offset();
+
+    float rToAxis    = FLT_MAX;
+    float xToOrigin  = FLT_MAX;
+    size_t iToAxis   = 0;
+    size_t iToOrigin = 0;
+
+    std::vector<const OctreeNode*> nodeStack;
+    nodeStack.push_back(m_rootNode.get());
+    while (!nodeStack.empty())
+    {
+        const OctreeNode* node = nodeStack.back();
+        nodeStack.pop_back();
+
+        if (!node->isLeaf())
+        {
+            for (int i = 0; i < 8; ++i)
+            {
+                OctreeNode* n = node->children[i];
+                if (n)
+                    nodeStack.push_back(n);
+            }
+            continue;
+        }
```

Comment:
This traversal seems to be out of the spirit of using a tree data structure: it's running `closestPoint` on *all* leaf nodes, rather than pruning early where possible based on the bounding boxes of the interior nodes.

Asymptotically this doesn't scale well.  It won't be noticeable with the really large leaf size I've got currently, but if the leaf size was made smaller in the future it might be a problem.  It seems a pity not to use the tree if it's available - I suspect you can just pull the first part of `closestPoint` out, and evaluate it for interior nodes.  If an interior node isn't in the path of the ray, you can discard it + all its children right away by not putting them onto the node traversal stack.